packages/nuxt/package.json

```diff
@@ -12,16 +12,28 @@
   "files": [
     "/build"
   ],
-  "main": "build/module.cjs",
-  "module": "build/module.mjs",
-  "types": "build/types.d.ts",
+  "main": "build/cjs/index.server.js",
+  "module": "build/esm/index.server.js",
+  "browser": "build/esm/index.client.js",
+  "types": "build/types/index.types.d.ts",
   "exports": {
+    "./package.json": "./package.json",
     ".": {
-      "types": "./build/types.d.ts",
-      "import": "./build/module.mjs",
-      "require": "./build/module.cjs"
+      "types": "./build/types/index.types.d.ts",
+      "browser": {
+        "import": "./build/esm/index.client.js",
+        "require": "./build/cjs/index.client.js"
+      },
+      "node": {
+        "import": "./build/esm/index.server.js",
+        "require": "./build/cjs/index.server.js"
+      }
     },
-    "./package.json": "./package.json"
+    "./module": {
+      "types": "./build/module/types.d.ts",
```

Comment:
TODO: Will we have to do the same as for solid here, where types for a submodule export have to be in the root? 😬 no need to adjust this in this PR, but have a look at it in a follow up, and maybe sync with @andreiborza  on this :)